### DISCUSSION TITLE
SPARQL Query to other SPARQL Endpoints

### DISCUSSION THREAD

**Discussion by WaltonG** [OpenRefine/SparqlExtension#11] [Q&A]
So far we can sucessfully send a SPARQL query to wikidata. Currently am working on sending a SPARQL query to any SPARQL enpoint. I was thinking of programmatically changing the below line with the endpoint url provided by the user.
https://github.com/OpenRefine/SparqlExtension/blob/b14a6a6c78f68b1b5f5a4103d6b09f7c1692d202/module/scripts/index/wikidata/Sparql.js#L8

The challenge am encountering is status code 307-internal redirect. I think its caused by 

https://github.com/OpenRefine/SparqlExtension/blob/b14a6a6c78f68b1b5f5a4103d6b09f7c1692d202/module/scripts/index/wikidata/Sparql.js#L95-L96

Am not sure how to edit the line so that it can accomodate any SPARQL endpoint

**Comment by Abbe98** [comment]
I don't think there is anything in the SPARQL standard(s) allowing one to check when the data was updated. Maybe there is an extension to the _SPARQL 1.1 Service Description_ or the _SPARQL 1.1 Protocol_ standards which expose such data but there is no way to use the _SPARQL 1.1 Query Language_ for such a check.

That said, isn't update lag a very Wikibase(or even Wikidata) specific issue?

**Reply by WaltonG** [reply]
Thanks for the heads up. Am getting somewhere as now am receiving status code 200

**Reply by ostephens** [reply]
What does the nocache parameter do? How does it compare to using a `cache-control: no-cache` header on the http request?

**Reply by WaltonG** [reply]
I think its used to avoid browser caching. However i have managed to query two enpoints https://lod.openlinksw.com/sparql and http://dbpedia.org/sparql by editing the below line

https://github.com/OpenRefine/SparqlExtension/blob/b14a6a6c78f68b1b5f5a4103d6b09f7c1692d202/module/scripts/index/wikidata/Sparql.js#L8

Users need to provide the correct SPARQL endpoint URI for the query to be successful.

As @Abbe98 stated i think the below code is not needed for a query to be successful

https://github.com/OpenRefine/SparqlExtension/blob/b14a6a6c78f68b1b5f5a4103d6b09f7c1692d202/module/scripts/index/wikidata/Sparql.js#L92-L127

**Reply by Abbe98** [reply]
Another thing that might be worth considering would be that if SPARQL queries are done from JavaScript/the client like above, endpoints not supporting CORS won't work and there will likely be SSL-related issues as well.

**Reply by WaltonG** [reply]
Sure ill have to do some enhancement and utilize the Openrefine HTTPClient
